homeassistant/components/matter/adapter.py

```diff
@@ -194,20 +195,35 @@ def _create_device_registry(
             identifiers.add((DOMAIN, f"{ID_TYPE_SERIAL}_{basic_info_serial_number}"))
             serial_number = basic_info_serial_number
 
-        model = (
-            get_clean_name(basic_info.productName) or device_type.__name__
+        # Model name is the human readable name of the model/product name
+        model_name = (
+            # productLabel is optional but preferred (e.g. Hue Bloom)
+            get_clean_name(basic_info.productLabel)
+            # alternative is the productName (e.g. LCT001)
+            or get_clean_name(basic_info.productName)
+            # if no product name, use the device type name
+            or device_type.__name__
             if device_type
             else None
         )
+        # Model ID is the non-human readable product ID
+        # we prefer the matter product ID so we can look it up in Matter DCL
+        if isinstance(basic_info, clusters.BridgedDeviceBasicInformation):
+            # On bridged devices, the productID is not available
+            # we use the product name as fallback model ID
+            model_id = get_clean_name(basic_info.productName)
```

Comment:
How will this fallback affect assumptions and use of the model id attribute in consumers? Are we sure it's better to have a fallback than to leave it empty?

Reply:
Can we add a test of a bridged device?

Reply:
Yeah, I had the same doubt. I actually pinged @balloob about it as well yesterday.

We collect modelID for the device database but in this case this is a bridged device. Maybe we shouldn't collect the modelID at all in that case because the only modelID we are interested in from a database perspective is the bridge itself (as that is the matter product).....

Reply:
Decided to go with None for bridged devices as it does not make any sense to have the modelId there.
As for adding a test for bridged devices; we do not have any tests now for bridges, we have a task on the backlog to add them. I now have a couple of  Matter bridges on my desk to test with so I'll setup the test foundations for matter bridges based on those in a separate PR. Lets first unblock this issue as its preventing the integration to start now.